tooling/minimize/src/rvalue.rs

```diff
@@ -54,6 +54,23 @@ impl<'cx, 'tcx> FnCtxt<'cx, 'tcx> {
                         rs::span_bug!(span, "Binary Op {op:?} not supported for type {lty_smir}."),
                 };
 
+                // cast boolean used in comparisons to Int to use them in IntRel ops
+                let (l, r) = match (bin_op, lty) {
```

Comment:
I think it's easier to check `(op, lty)` against `(BinOp::IntRel(_), Type::Bool)`.